### PR TITLE
mesh-config-json: fix mesh node configuration incompatibility

### PR DESCRIPTION
**Pull request by przemyslawfierek-silvair**
This patch fixes problem with Mesh Node configuration incompatibility.
Previous configuration format keeped keys indexes as hexstrings that was
changned meanwhile to the int format, but migration path has not been
implemented.

### CODE REVIEW

File: mesh/mesh-config-json.c [hunk 1]
```diff
@@ -86,13 +86,47 @@ static bool save_config(json_object *jnode, const char *fname)
...
+			if (errno == EINVAL)
+				return false;
+
+			break;
+
+		case json_type_string: {
+			uint16_t hexstr_value;
+			const char* str;
+
+			str = json_object_get_string(jvalue);
```
Comment (michallowasrzechonek-silvair) [inline mesh/mesh-config-json.c]: Can't you just try to read string, and when that fails, read an int, and when that fails return error?

I don't think you need to switch by type

Comment (przemyslawfierek-silvair) [reply mesh/mesh-config-json.c]: I've made a quick test and it doesn't work.

Comment (przemyslawfierek-silvair) [reply mesh/mesh-config-json.c]:
```
{
      "index":10,
      "key":"deadbeef882491c9407b4fbf2fee933",
      "keyRefresh":0
}
```
=> index == 16 (not 10) ;/

Comment (michallowasrzechonek-silvair) [reply mesh/mesh-config-json.c]: Ok, docs say it returns the string representation even if value isn't one. So we need to check the type. Still, I'd drop the switch and use a plain `if`.

Comment (przemyslawfierek-silvair) [reply mesh/mesh-config-json.c]: Check out this version, it should be ok.

**Review by michallowasrzechonek-silvair — APPROVED** [review]

**Pull request merged**